api/plugin/file/fileutils.go

```diff
@@ -63,3 +84,18 @@ func deleteFileAndParentDirIfNeeded(path string) {
 		dir = filepath.Dir(dir)
 	}
 }
+
+// Check whether a directory exists and is valid.
+func isValidDirectory(dirPath string) bool {
```

Comment:
There is a test of the error to see if it is a "file or path does not exist".  I think it's part of the os package.  If the path returns an error and the error is NOT file does not exist, then you at least need to be logging that.